NuRadioMC/simulation/simulation.py

```diff
@@ -401,752 +1339,325 @@ def __init__(self, inputfilename,
-                            if self._cfg['propagation']['module'] == 'radiopropa':
-                                logger.error('Presimulation can not be used with the radiopropa ray tracer module')
-                                raise Exception('Presimulation can not be used with the radiopropa ray tracer module')
-                            sg_pre = self._fin_stations["station_{:d}".format(self._station_id)]
-                            ray_tracing_solution = {}
-                            for output_parameter in self._raytracer.get_output_parameters():
-                                ray_tracing_solution[output_parameter['name']] = sg_pre[output_parameter['name']][self._shower_index, channel_id]
-                            self._raytracer.set_solution(ray_tracing_solution)
+                # group events into events based on signal arrival times
+                events = group_into_events(station, event_group, particle_mode, self._config['split_event_time_diff'], time_logger=self.__time_logger)
+
+                evt_group_triggered = False
+                for evt in events:
+                    station = evt.get_station()
+                    apply_det_response(evt, self._det, self._config, self.detector_simulation_filter_amp,
+                                       bool(self._config['noise']),
+                                       self._Vrms_efield_per_channel, self._integrated_channel_response,
+                                       self._noiseless_channels, 
+                                       time_logger=self.__time_logger,
+                                       channel_ids=channel_ids,
+                                       detector_simulation_part2=self.detector_simulation_part2)
+
+                    # calculate trigger
+                    self.__time_logger.start_time("trigger")
+                    self.detector_simulation_trigger(evt, station, self._det)
+                    logger.debug(f"event {evt.get_run_number()},{evt.get_id()} tiggered: {evt.get_station().has_triggered()}")
+                    self.__time_logger.stop_time("trigger")
+                    if not evt.get_station().has_triggered():
+                        continue
+                    triggerTimeAdjuster.run(evt, station, self._det)
+                    evt_group_triggered = True
+                    output_buffer[sid][evt.get_id()] = evt
+                # end event loop
+                # now simulate non-trigger channels
+                # we loop through all non-trigger channels and simulate the electric fields for all showers. 
+                # then we apply the detector response to the electric fields and find the event in which they will be visible in the readout window
+                non_trigger_channels = list(set(self._det.get_channel_ids(sid)) - set(channel_ids))
+                if (len(non_trigger_channels) > 0):
```

Comment:
By default we only want to simulate non-triggered channels, right? How is that enforced? Looks like we simulate them anyway but than just not add them to the events/stations.

Reply:
we always simulate them if we have a trigger. The big difference to before is that we don't calculated all channels for events that don't trigger.

Reply:
sure but this is not the case at the moment

Reply:
with the `continue`
```
                if not evt.get_station().has_triggered():
                        continue

                    triggerTimeAdjuster.run(evt, station, self._det)
                    evt_group_triggered = True
                    output_buffer[sid][evt.get_id()] = evt

```
we only skip the next 3 lines in the for-loop `for evt in events:`. The block of `if (len(non_trigger_channels) > 0):` is still executed